Cdn check

petebacondarwin opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

chores

**What is the current behavior? (You can also link to an open issue here)**
- failure to collect the correct CDN version information can cause the docs website to crash
- builds always attempt to make requests to servers, even if you are offline

**What is the new behavior (if this is a feature change)?**
- builds fail if the CDN version information is missing or if the production version of the app is not working
- you can set your local builds to run without making requests to the server

**Does this PR introduce a breaking change?**

Nope

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

Related to #14761

petebacondarwin committed:
chore(Gruntfile): kill the build if the CDN version was not found

petebacondarwin committed:
chore(app.scenario): test docs app against production libraries

We have had issues where the docs application has gone down because
our build included invalid URLs to the production angular libraries.

This change runs a subset of the docs e2e tests directly against these
production libraries to ensure that they are accessible. If not then these
tests will fail and the CI build will abort, preventing the docs app from
being updated.

petebacondarwin committed:
chore(version-info): add offline-build support

Previously it was slow and problematic to run builds if you were not connected
to the internet because the build scripts make requests to github and the
Google CDN to find out version information.

You can now disable these requests by setting the ANGULAR_1_OFFLINE_BUILD
environment variable to a truthy value.

Narretz commented:
OFFLINE is not 100% correct, it's more like NO_REMOTE_REQUESTS

gkalpak commented on docs/app/e2e/app.scenario.js:
```diff
@@ -39,7 +39,7 @@ describe('docs.angularjs.org', function () {
 
     it('should change the page content when clicking a link to a service', function () {
-      browser.get('build/docs/index.html');
+      browser.get('build/docs/index-production.html');
```
So, by using `-production` we are not testing against our local build, but only against the CDN version, right?
This makes sense, since it is a separate app that just happens to live on the same repo and the app will use a CDN version (not our local build).

petebacondarwin replied:
Yes, so there are other tests in this folder that hit the normal `index.html`.
Also all the main e2e tests from the examples are run against local angular.

gkalpak commented on lib/versions/version-info.js:
```diff
@@ -138,6 +141,10 @@ var getCdnVersion = function() {
       if (!cdnVersion) {
+        if (OFFLINE) {
+          // We are offline so just use the most recent version
+          cdnVersion = version;
```
Shouldn't we return here?

Narretz replied:
Yes, otherwise we are making one request

petebacondarwin replied:
OK!

gkalpak commented on lib/versions/version-info.js:
```diff
@@ -204,3 +211,13 @@ exports.gitRepoInfo = gitRepoInfo = getGitRepoInfo();
+if (OFFLINE) {
+  console.log('==============================================================================================');
+  console.log('Running Offline:');
+  console.log(' - this due to the "ANGULAR_1_OFFLINE_BUILD" environment variable being defined.');
```
_this due_  -->  _this is due_ (?)

petebacondarwin replied:
Right

petebacondarwin commented:
OK @Narretz, I can change that constant. Any other concerns?

Narretz commented:
I'm still testing.

petebacondarwin committed:
chore(version-info): add offline-build support

Previously it was slow and problematic to run builds if you were not connected
to the internet because the build scripts make requests to github and the
Google CDN to find out version information.

You can now disable these requests by setting the NG1_BUILD_NO_REMOTE_VERSION_REQUESTS
environment variable to a non-empty value

Narretz commented on lib/versions/version-info.js:
```diff
@@ -204,3 +213,13 @@ exports.gitRepoInfo = gitRepoInfo = getGitRepoInfo();
+
+if (NO_REMOTE_REQUESTS) {
+  console.log('==============================================================================================');
+  console.log('Running Offline:');
```
"Running without getting remote tag and cdn version data"?

petebacondarwin replied:
"Running with no remote requests for version data"?

petebacondarwin committed:
chore(version-info): add offline-build support

Previously it was slow and problematic to run builds if you were not connected
to the internet because the build scripts make requests to github and the
Google CDN to find out version information.

You can now disable these requests by setting the NG1_BUILD_NO_REMOTE_VERSION_REQUESTS
environment variable to a non-empty value

Narretz commented:
LGTM. Please be prepared for a follow-up that tries to make the "online" version work on Windows (because it doesn't like `-o dev/null`)
